### ISSUE TITLE
Fix ImGui::NewFrame throws assertion failure in VM with VSYNC+waiting disabled

### ISSUE THREAD

**Issue by adamkewley** [ComputationalBiomechanicsLab/opensim-creator#617]
Because (I believe) the VM's performance counters aren't incrementing properly, which causes a timedelta of `0` to happen, which causes ImGui to to haywire and throw an assertion:

```text
osc: /home/adam/Desktop/opensim-creator/third_party/imgui/imgui.cpp:9324: void ImGui::ErrorCheckNewFrameSanityChecks(): Assertion `(g.IO.DeltaTime > 0.0f || g.FrameCount == 0) && "Need a positive DeltaTime!"' failed.

osc: critical error: signal 6 (Aborted) received from OS: address is 0x3e800000af6 from 0x7f165c9b3a7c

    #1  /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c) [0x7f165c9b3a7c]

    #2  /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f165c95f520]

    #3  /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c) [0x7f165c9b3a7c]

    #4  /lib/x86_64-linux-gnu/libc.so.6(raise+0x16) [0x7f165c95f476]

    #5  /lib/x86_64-linux-gnu/libc.so.6(abort+0xd3) [0x7f165c9457f3]

    #6  /lib/x86_64-linux-gnu/libc.so.6(+0x2871b) [0x7f165c94571b]

    #7  /lib/x86_64-linux-gnu/libc.so.6(+0x39e96) [0x7f165c956e96]

    #8  ./osc(_ZN5ImGui8NewFrameEv+0x387) [0x55a543e72107]

    #9  ./osc(_ZN3osc12MainUIScreen4Impl6onDrawEv+0x308) [0x55a5433c2608]

    #10 ./osc(_ZN3osc3App4Impl17mainLoopUnguardedEv+0xfa9) [0x55a54376c5c9]

    #11 ./osc(_ZN3osc3App4Impl4showESt10unique_ptrINS_6ScreenESt14default_deleteIS3_EE+0x34a) [0x55a543771c3a]

    #12 ./osc(_ZN3osc3App4showESt10unique_ptrINS_6ScreenESt14default_deleteIS2_EE+0xe3) [0x55a54375bb03]

    #13 ./osc(main+0x5ad) [0x55a543391efd]

    #14 /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f165c946d90]

    #15 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f165c946e40]

    #16 ./osc(_start+0x25) [0x55a54339caa5]
```

**Comment by adamkewley** [comment]
After a bit of digging around with "this should never happen"-style assertions I found that the bug was because `SDL_GetPerformanceCounter` can sometimes return the same value twice, rather than monotonically increasing. This then later causes a DeltaTime of `0` to slip into the backend and trigger the assertion.

I wrote a quick hotfix for this that adds a single tick to the performance counter whenever this happens, so that, from ImGui's perspective, time has moved forward (by an insanely small amount) and, when the counter starts increasing again, the mitigation is ignored.

Upstream here: https://github.com/ocornut/imgui/pull/6189
